src/main/java/com/faforever/client/preferences/ui/SettingsController.java

```diff
@@ -255,9 +286,29 @@ private void configureThemeSelection(Preferences preferences) {
     );
   }
 
-  private void configureLanguageSelection() {
-    languageComboBox.setItems(FXCollections.singletonObservableList("English"));
-    languageComboBox.getSelectionModel().select(0);
+  private void configureLanguageSelection(Preferences preferences) {
+    languageComboBox.setItems(FXCollections.observableArrayList(LanguageInfo.getAllDisplayNames()));
```

Comment:
Thanks for that tip was really effective and removed near by evry code in TimeInfo and LanguageInfo... And all the previous changes were senseless :D